src/dpp/httpsclient.cpp

```diff
@@ -103,7 +103,7 @@ multipart_content https_client::build_multipart(const std::string &json, const s
 			/* Multiple files */
 			for (size_t i = 0; i < filenames.size(); ++i) {
 				content += part_start + "name=\"files[" + std::to_string(i) + "]\"; filename=\"" + filenames[i] + "\"";
-				content += "\r\nContent-Type: " + (mimetypes.size() < i || mimetypes[i].empty() ? default_mime_type : mimetypes[i]) + two_cr;
+				content += "\r\nContent-Type: " + (mimetypes.size() <= i || mimetypes[i].empty() ? default_mime_type : mimetypes[i]) + two_cr;
```

Comment:
wait.... no.... this condition is just nasty. should it be refactored to be like i < mimetypes.size()?

Reply:
It could be, like
```cpp
				content += "\r\nContent-Type: " + (i < mimetypes.size() && !mimetypes[i].empty() ? mimetypes[i] : default_mime_type) + two_cr;
```
but it doesn't make much of a difference, still a nasty condition.